docs/contribution_guide/fairlearn_repository_overview.rst

```diff
@@ -0,0 +1,8 @@
+Overview of the Fairlearn repository
```

Comment:
I don't think we need this part. Why not include the two subparts directly into `contribution_guide/index.rst`?

Reply:
I had misunderstood something about how toctree works. Changed this accordingly.